### PR TITLE
more 7.2 improvements

### PR DESCRIPTION
**Pull request by jtwaleson**

### CODE REVIEW

**Comment by knorrie** [comment]
Ah, grmbl, I'll push the import removal down into the 'Python 3 compatibility' commit, thanks.

File: src/m2ee/config.py [hunk 1]
```diff
@@ -89,25 +89,20 @@ def _setup_classpath(self):
 
     def _try_load_json(self, jsonfile):
         logger.debug("Loading json configuration from %s" % jsonfile)
-        fd = None
         try:
-            fd = open(jsonfile)
+            with open(jsonfile) as fd:
```
Comment (knorrie) [inline src/m2ee/config.py]: I noticed the usage of fd for a file object earlier (wrote that myself in the past), but didn't fix it yet. As it obviously sets a wrong example, we should now.

If i see fd as variable, I expect a simple integer number which is a file descriptor, and nothing else.

Can you please s/fd/f/ in all those places?

File: src/m2ee/config.py [hunk 2]
```diff
@@ -89,25 +89,20 @@ def _setup_classpath(self):
...
         except Exception as e:
             logger.debug("Error reading configuration file %s: %s; "
                          "ignoring..." % (jsonfile, e))
             return {}
-
-        config = None
-        try:
-            config = json.load(fd)
-        except Exception as e:
+        except ValueError as e:
```
Comment (knorrie) [inline src/m2ee/config.py]: If you want the valuerror code to ever be reached, it has to go before except Exception, which eats everything.

Comment (jtwaleson) [reply src/m2ee/config.py]: Oops, I forgot to replace that Exception with IOError. Fixed and rebased.

File: src/m2ee/config.py (hunk 1 above)
Comment (jtwaleson) [reply src/m2ee/config.py]: Done.
`ack fd -l | xargs sed -i 's/fd/f/'`
`ack close_fs -l | xargs sed -i 's/close_fs/close_fds/'`

**Comment by knorrie** [comment]
Oh, haha, github can't handle heavy rebase -i usage...

I merged the isinstance and import change down to the python 3 commits and picked the last version of your with path. Thanks.

**Comment by knorrie** [comment]
Done!

**Pull request closed without merging**